modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -257,39 +257,36 @@ function DataSetDisplay({
 		}
 	}, [wrapperRef]);
 
-	const refreshData = useCallback(
-		(successNotification) => {
-			setLoading(true);
-
-			return requestData()
-				.then((data) => {
-					const {
-						message,
-						showSuccessNotification,
-					} = successNotification;
-
-					if (showSuccessNotification) {
-						openToast({
-							message:
-								message ||
-								Liferay.Language.get('table-data-updated'),
-							type: 'success',
-						});
-					}
+	function refreshData(successNotification) {
```

Comment:
Fabio talked to me on slack and said he was having problems with the refresh method getting called but not updating the UI. This was the way hw found to work around that. I didn't fully understand why this was happening, but I guess this workaround could't be harmful. I'm not forwarding right now to give a chance for other people to chime in.

@FabioDiegoMastrorilli If this is urgent, let me know and I'll forward right away.

/cc @wincent

Reply:
Based on a superficial look at this, it's not obvious why the UI wouldn't refresh before, but it does after:

- Before: `refreshData` is memoized using `useCallback`
- After: `refreshData` is just a normal function.

Just guessing but... In this commit it's not obvious how that would make a difference, but because `refreshData` is passed around a lot, it is possible that the use of `useCallback` was leading the function to have a stable identity which was in turn causing some children's `useEffect` callbacks _not_ to run.

So, I suspect that this fixes the problem at the cost of potentially introducing some additional re-renders that might not be necessary (in addition to the one that _is_ necessary).

If you wanted to analyze this further, I think it involve looking at why `updateDataSetItems` isn't having the expected side-effects of forcing a re-render.

Reply:
The fetch via `requestData` gets performed and return the right data but `updateDataSetItems` doesn't seem working. 

The thing with the data set display is that sometimes we need to update it even when nothing changes in it (I can delete an element via dropdown actions or I can update it via side panel) and `refreshData` still has to work even if the component state doesn't change.

My assumption was that memoized functions could jeopardised these refreshes so I got rid of one which wasn't executing expensive operations.